source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm

```diff
@@ -76,11 +77,14 @@ - (UIView *)renderButtons:(ACRView *)rootView
     ACOBaseActionElement *acoElem = [[ACOBaseActionElement alloc] init];
     ACRContentHoldingUIScrollView *containingView = [[ACRContentHoldingUIScrollView alloc] init];
     [superview addArrangedSubview:containingView];
-    float accumulatedWidth = 0, accumulatedHeight = 0, spacing = [config getHostConfig] -> GetActions().buttonSpacing, maxWidth = 0, maxHeight = 0;
+    float accumulatedWidth = 0, accumulatedHeight = 0, spacing = adaptiveActionConfig.buttonSpacing, maxWidth = 0, maxHeight = 0;
     if (elems.empty()) {
         return containingView;
     }
-    for (const auto &elem : elems) {
+
+    unsigned long uMaxActionsToRender = MIN(adaptiveActionConfig.maxActions, elems.size());
+    for (auto i = 0; i < uMaxActionsToRender; i++) {
```

Comment:
>for (auto i = 0; i < uMaxActionsToRender; i++) [](start = 4, length = 46)

Should we add a warning if we have more than the max actions?  Just checked and we do in UWP.

Reply:
agree with @RebeccaAnne here :)

---
In reply to: [363433129](https://github.com/microsoft/AdaptiveCards/pull/3696#discussion_r363433129) [](ancestors = 363433129)

Reply:
@nesalang we should also have unit tests for this.